### PR TITLE
Landing examples

### PR DESCRIPTION
**Pull request by sgrove**

### CODE REVIEW

File: components/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
+// @flow
```
Comment (dwwoelfel) [inline components/SignUp.js]: remove?

File: src/Landing.js [hunk 2]
```diff
@@ -147,60 +177,212 @@ class Landing extends React.Component<Props, State> {
...
+                  alt="OneGraph"
+                  style={{float: 'left', margin: '-15px'}}
+                />
+
+                <h1 style={{display: 'inline-block', fontSize: '40px'}}>
+                  OneGraph
+                </h1>
+              </div>
+
+              <h3 style={{fontSize: '30pt'}}>
```
Comment (dwwoelfel) [inline src/Landing.js]: pt?

File: src/Landing.js [hunk 3]
```diff
@@ -147,60 +177,212 @@ class Landing extends React.Component<Props, State> {
...
+            <Col md={24} style={{textAlign: 'left', marginBottom: 10}}>
+              <h1>
+                "Find the <strong>5 most recent charges</strong> on{' '}
+                <strong>Stripe</strong>"
+              </h1>
             </Col>
           </Row>
+          <Row>
+            <Col>
+              <MiniOneGraphiQL2
```
Comment (dwwoelfel) [inline src/Landing.js]: Can create a separate component to make it easier to follow. Would also be good to define the query with a separate var (or, even better, bundle it with its result), to prevent the weird formatting with the template string.

File: src/MiniOneGraphiQL2.js [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+  isListType,
+} from 'graphql';
+
+import 'graphiql/graphiql.css';
+
+import type {OneGraphAuth} from 'onegraph-auth';
+
+import schemaSDL from './graphqlSchema';
+
+var Fakerator = require('fakerator');
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: const

File: src/MiniOneGraphiQL2.js [hunk 5]
```diff
@@ -0,0 +1,379 @@
+import Config from './Config';
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Can we give this a more descriptive name over MiniOneGraphiQL2?

File: src/MiniOneGraphiQL2.js [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+
+const schema = buildSchema(schemaSDL);
+
+type Props = {|
+  isLoggedIn: boolean,
+  stripeLoginButton: React.Node,
+  oneGraphAuth: OneGraphAuth,
+  query: string,
+  result: string,
+  schema: ?GraphQLSchema,
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Why take schema as a prop? Could you just always use the schema defined in this file?

File: src/MiniOneGraphiQL2.js [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+import 'graphiql/graphiql.css';
+
+import type {OneGraphAuth} from 'onegraph-auth';
+
+import schemaSDL from './graphqlSchema';
+
+var Fakerator = require('fakerator');
+var fakerator = Fakerator();
+
+const schema = buildSchema(schemaSDL);
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Can we run this only if the component is mounted? Probably don't want to add the extra cpu cycles for people who are skipping the landing page and going to the dashboard/pricing/etc.

File: src/MiniOneGraphiQL2.js [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+
+const getGenerator = (gqlObjTypeName, fieldName, fieldType) => {
+  return (
+    generators[gqlObjTypeName + capitalize(fieldName)] ||
+    generators[fieldName] ||
+    generators[fieldType]
+  );
+};
+
+export function fetchQuery(query: string): Promise<Object> {
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: What are we using fetchQuery for?

File: src/MiniOneGraphiQL2.js [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+    const randType = types[Math.floor(Math.random() * types.length)];
+    return {type: randType, isList: isList};
+  } else if (isListType(gqlType)) {
+    return findBaseGraphQLType(gqlType.ofType, true);
+  } else {
+    return findBaseGraphQLType(gqlType.ofType, isList);
+  }
+};
+
+var handler = {
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: const

Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Could we get a comment describing what this thing does?

File: src/MiniOneGraphiQL2.js [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+      const value = obj[prop];
+      const isObject = typeof value === 'object';
+      const returnValue = isObject ? new Proxy(value, handler) : value;
+
+      return returnValue;
+    } else {
+      const currentObjTypeName = obj.graphQLTypeName;
+      const graphQLType = schema.getType(currentObjTypeName);
+      var fields;
+      if (isObjectType(graphQLType)) {
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: isObjectType(graphQLType) || isInterfaceType(graphQLType)

Comment (dwwoelfel) [reply src/MiniOneGraphiQL2.js]: What happens if it's a non-null type?

File: src/MiniOneGraphiQL2.js [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+        fields = graphQLType.getFields();
+      } else if (isInterfaceType(graphQLType)) {
+        fields = graphQLType.getFields();
+      } else {
+        fields = null;
+      }
+      const lookupFieldExists = !!(fields && fields[prop]);
+
+      if (lookupFieldExists) {
+        const fullBaseType = findBaseGraphQLType(fields[prop].type);
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: What does `full` indicate here?

File: src/MiniOneGraphiQL2.js [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+        const baseType = fullBaseType.type;
+        const isList = !!fullBaseType.isList;
+
+        const nextGraphQLType = schema.getType(baseType.name);
+
+        var returnValue;
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Why have this var instead of `return`ing?

Should also be `let`

File: src/MiniOneGraphiQL2.js [hunk 13]
```diff
@@ -0,0 +1,379 @@
...
+        if (isScalarType(nextGraphQLType)) {
+          if (nextGraphQLType.name === 'String') {
+            const generator = getGenerator(
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Looks like you can define generator in the else.

File: src/MiniOneGraphiQL2.js [hunk 14]
```diff
@@ -0,0 +1,379 @@
...
+              nextGraphQLType.name,
+            );
+
+            if (isList) {
+              returnValue = new Array(
+                Math.max(1, Math.floor(Math.random() * 2)),
+              )
+                .fill(null)
+                .map(() => {
+                  return getGenerator(
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: I had a hard time seeing that it was calling the generator. Could you do:

```
const generator = getGeneragor(...);
return generator();
```

File: src/MiniOneGraphiQL2.js [hunk 15]
```diff
@@ -0,0 +1,379 @@
...
+                    currentObjTypeName,
+                    prop,
+                    nextGraphQLType.name,
+                  )();
+                });
+            } else {
+              returnValue = generator();
+            }
+            return returnValue;
+          } else if (nextGraphQLType.name === 'Int') {
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Why the special check for `Int`?

File: src/MiniOneGraphiQL2.js [hunk 16]
```diff
@@ -0,0 +1,379 @@
...
+    return {type: randType, isList: isList};
+  } else if (isListType(gqlType)) {
+    return findBaseGraphQLType(gqlType.ofType, true);
+  } else {
+    return findBaseGraphQLType(gqlType.ofType, isList);
+  }
+};
+
+var handler = {
+  get: function(obj, prop) {
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Could you break this up into smaller functions? It's a bit hard to follow because it doesn't fit on a single page.

File: src/MiniOneGraphiQL2.js [hunk 17]
```diff
@@ -0,0 +1,379 @@
...
+          return returnValue;
+        }
+      } else {
+        return null;
+      }
+    }
+  },
+};
+
+var source = new Proxy({graphQLTypeName: 'Query'}, handler);
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: const

File: src/MiniOneGraphiQL2.js [hunk 18]
```diff
@@ -0,0 +1,379 @@
...
+    this.setState({query});
+  };
+
+  _handleRunQuery = () => {
+    this.setState({result: '', fetching: true});
+    setTimeout(() => {
+      graphql(
+        schema,
+        this.state.query,
+        // this.state.resultSource
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: remove comment?

File: src/MiniOneGraphiQL2.js [hunk 19]
```diff
@@ -0,0 +1,379 @@
...
+        source,
+      ).then(result => {
+        // Prints
+        // {
+        //   data: { hello: "world" }
+        // }
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: Remove comment?

File: src/MiniOneGraphiQL2.js [hunk 20]
```diff
@@ -0,0 +1,379 @@
...
+    this.setState({showStripeConnectOverlay: false});
+  };
+
+  componentDidMount() {
+    this.setState(state => {
+      return {
+        ...state,
+        schema: this.props.schema,
+        query: this.props.query,
+
```
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: empty line

File: src/MiniOneGraphiQL2.js (hunk 5 above)
Comment (dwwoelfel) [inline src/MiniOneGraphiQL2.js]: It would be really nice to make this use MiniOneGraphiQL. It's going to be really hard to keep these two in sync.

File: src/graphqlSchema.js [hunk 21]
```diff
@@ -0,0 +1,14312 @@
+let schema = `#  null
```
Comment (dwwoelfel) [inline src/graphqlSchema.js]: This is really big! Can you do a `yarn build` to see how large the compiled js file is?

**Comment by sgrove** [comment]
@dwwoelfel lgtm?
